quesma/schema/registry.go

```diff
@@ -39,6 +39,14 @@ type (
 func (s *schemaRegistry) loadSchemas() (map[TableName]Schema, error) {
 	definitions := s.dataSourceTableProvider.TableDefinitions()
 	schemas := make(map[TableName]Schema)
+	if len(*s.indexConfiguration) == 0 { // no index configs => table auto-discovery is enabled
```

Comment:
In the near future, I plan to change the way auto discovery is enabled from "no index configs => auto discovery" to:
```yaml
# config of query processor:
  "*": # syntax not final
    "target": ["clickhouse"]
  ...
  "other_index": ...
```

That way you could mix autodiscovery and explicit index configs.

Maybe this is not necessary in this PR, but you could improve the logic to be able to merge `s.indexConfiguration` and definitions not present in `s.indexConfiguration`.

Reply:
When working on this further I noticed  that `len(*s.indexConfiguration) == 0` became a recurring pattern so I'll move it to dedicated function, which will also make the code more bulletproof

Reply:
Nice!